src/Puzzle.java

```diff
@@ -7,21 +11,135 @@
     private final static int PANGRAM_BONUS = 7;
 
     /** The primary (required) letter of the puzzle. */
     private char primaryLetter;
-
     /** The secondary letters of the puzzle. */
     private char[] secondaryLetters;
-
+    /** The list of all valid words for the puzzle. */
     private ArrayList<String> validWords;
+    /** The list of all words currently found in the puzzle. */
     private ArrayList<String> foundWords;
+    /** The total number of points that can be earned in the puzzle. */
     private int totalPoints;
+    /** The number of points currently earned in the puzzle. */
     private int earnedPoints;
-    private int[] rankPoints;
+
+    /**
+     * Constructs a Puzzle object from the required letter, and the six other
+     * acceptable letters. Fills validWords by parcing through dictionaryFile.
+     * 
+     * @param primaryLetter The required letter for the puzzle
+     * @param secondaryLetters The six other acceptable letters for the puzzle
+     * @param dictionaryFile The dictionary file to be used to generate validWords.
+     * @throws IOException if an I/O error occurs.
+     */
+    public Puzzle(char primaryLetter, char[] secondaryLetters,
+                  FileReader dictionaryFile) throws IOException {
+        this.primaryLetter = primaryLetter;
+        this.secondaryLetters = Arrays.copyOf(secondaryLetters,
+                                              secondaryLetters.length);
+        this.validWords = new ArrayList<String>();
+        this.foundWords = new ArrayList<String>();
+        this.totalPoints = 0;
+        this.earnedPoints = 0;
+
+        BufferedReader bufferedReader = new BufferedReader(dictionaryFile);
+
+        dictLoop:
+        for (String word = bufferedReader.readLine(); word != null; 
+             word = bufferedReader.readLine()) {
+            if (word.indexOf(primaryLetter) == -1) {
+                continue;
+            }
+
+            for (char c : secondaryLetters) {
+                if (word.indexOf(c) == -1) {
+                    continue dictLoop;
+                }
+            }
+
+            // To get to this point, all letters of the puzzle are in the word
+            this.validWords.add(word);
+            this.totalPoints += this.wordValue(word);
+        }
+
+        bufferedReader.close();
+    }
+
+    /**
+     * Create a puzzle from a given starting word. Fills validWords by parcing
+     * through dictionaryFile.
+     * 
+     * @param word The starting word for the puzzle
+     * @param dictionaryFile The dictionary file to be used to generate validWords.
+     * @return A puzzle based on the starting word
+     * @throws IllegalArgumentException if the word is not usable as a starting
+     *                                  word
+     * @throws IOException if an I/O error occurs.
+     */
+    public static Puzzle fromWord(String word, FileReader dictionaryFile) 
+            throws IllegalArgumentException, IOException {
```

Comment:
Maybe the starting word length requirement should be a constant?

Reply:
I'll lightly refactor that and some other magic numbers out. Good idea.